### PR TITLE
MP OpenAPI - More test cases for complex/negative/corner scenarios

### PR DESCRIPTION
**Pull request by fabiobrz**
* Implemented test cases for static files: multiple, big and non-valid static file content
* Implemented test case to cover usage of core and extension properties to disable subsystem/annotations scanning
* Implemented test case for context root/server information

Furthermore, `AcceptHeadersAndFormatParameterInteractionTest` has been fixed according to last DEV commit.

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- N/A Link to the passing job is provided: no job can execute the module
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

File: microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/HybridDocumentationTest.java [hunk 1]
```diff
@@ -81,7 +81,6 @@
     @Deployment(name = PROVIDER_DEPLOYMENT_NAME, order = 1, testable = false)
     public static Archive<?> serviceProviderDeployment() {
         //  MP OpenAPI disabled on this Services Provider deployment for testing purposes, we don't want it here
```
Comment (zhantemirov) [inline microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/HybridDocumentationTest.java]: Since the property was moved to line 94, could this commentary also moved before line 94 in order to have a clear explanation why we disabling this property?

File: microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/misc/ServerContextRootTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                        ProviderApplication.class,
+                        District.class,
+                        DistrictEntity.class,
+                        DistrictService.class,
+                        InMemoryDistrictService.class,
+                        DistrictsResource.class,
+                        RoutingServiceConstants.class,
+                        EchoResource.class)
+                .addAsManifestResource(new StringAsset("mp.openapi.extensions.enabled=false"), "microprofile-config.properties")
+                //.addAsManifestResource(new StringAsset("mp.openapi.scan.disable=true"), "microprofile-config.properties")
```
Comment (zhantemirov) [inline microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/misc/ServerContextRootTest.java]: I suppose this could be removed.

**Comment by zhantemirov** [comment]
Besides having a couple of notes, it LGTM.

File: microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/HybridDocumentationTest.java (hunk 1 above)
Comment (fabiobrz) [reply microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/HybridDocumentationTest.java]: Yes, definitely, thanks.

File: microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/misc/ServerContextRootTest.java (hunk 2 above)
Comment (fabiobrz) [reply microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/misc/ServerContextRootTest.java]: Sure, thanks for catching this.

**Comment by fabiobrz** [comment]
Hi @zhantemirov I pushed changes to address your comments. For the first one I decided to add Javadoc to the method itself. Let me know whether you're ok with it.

File: microprofile-open-api/src/test/resources/META-INF/openapi-broken.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+#  schemas:
+#    DistrictEntity:
+#      type: object
+#      properties:
+#        code:
+#          type: string
+#        name:
+#          type: string
+#        obsolete:
+#          type: boolean
```
Comment (mnovak1) [inline microprofile-open-api/src/test/resources/META-INF/openapi-broken.yaml]: I think those can be removed.

**Comment by mnovak1** [comment]
@fabiobrz just minor thing, otherwise LGTM.

**Comment by fabiobrz** [comment]
Hi @mnovak1, thanks. I pushed a fix to the yaml following your last comment.

**Comment by mnovak1** [comment]
thanks, merged
